tests/model_serving/model_server/llmd/utils.py

```diff
@@ -187,3 +193,242 @@ def get_llmd_pods():
+        dyn_client=client,
+        namespace=llmisvc.namespace,
+        label_selector=(
+            f"{Pod.ApiGroup.APP_KUBERNETES_IO}/part-of=llminferenceservice,"
+            f"{Pod.ApiGroup.APP_KUBERNETES_IO}/name={llmisvc.name}"
+        ),
+    ):
+        labels = pod.instance.metadata.get("labels", {})
+        if labels.get(f"{Pod.ApiGroup.APP_KUBERNETES_IO}/component") == "llminferenceservice-router-scheduler":
+            return pod
+    return None
+
+
+def send_prefix_cache_test_requests(
+    llmisvc: LLMInferenceService,
+    token: str,
+    num_requests: int = 20,
+) -> int:
+    """
+    Send N identical requests to validate prefix cache.
+
+    This function sends the same prompt multiple times to test cache affinity.
+    All requests after the first should hit the cache and route to the same pod.
+
+    Args:
+        llmisvc: The LLMInferenceService to send requests to
+        token: Authentication token
+        num_requests: Number of identical requests to send (default 20)
+
+    Returns:
+        int: Number of successful requests completed
+    """
+    successful_requests = 0
+    failed_requests = 0
+
+    # Single prompt to be cached
+    cached_prompt = (
+        "Explain in detail the fundamental principles of quantum mechanics including "
+        "wave-particle duality, superposition, and entanglement in simple terms. "
+        "Additionally, describe how these quantum phenomena differ from classical physics "
+        "and why they are important for understanding the nature of reality at the atomic scale."
+    )
+
+    LOGGER.info(f"Sending {num_requests} identical requests to test prefix cache")
+
+    for i in range(num_requests):
+        LOGGER.info(f"Sending request {i + 1}/{num_requests}")
+        inference_config = {
+            "default_query_model": {
+                "query_input": cached_prompt,
+                "query_output": r".*",
+                "use_regex": True,
+            },
+            "chat_completions": TINYLLAMA_INFERENCE_CONFIG["chat_completions"],
+        }
+
+        try:
+            verify_inference_response_llmd(
+                llm_service=llmisvc,
+                inference_config=inference_config,
+                inference_type="chat_completions",
+                protocol=Protocols.HTTPS,
+                use_default_query=True,
+                insecure=False,
+                model_name=llmisvc.instance.spec.model.name,
+                token=token,
+                authorized_user=True,
+            )
+            successful_requests += 1
+        except Exception as e:
+            LOGGER.error(f"Request {i + 1} failed: {e}")
+            failed_requests += 1
+
+    # Log statistics
+    if failed_requests > 0:
+        LOGGER.warning(f"Step 1: {successful_requests}/{num_requests} requests completed, {failed_requests} failed")
+    else:
+        LOGGER.info(f"Step 1: {successful_requests}/{num_requests} requests completed successfully")
```

Comment:
This does not seem to add any extra value and seems to be no-op. Can we please remove it? You already printed to console, which request failed.